lib/elasticsearch/elastic-query-builder.js

```diff
@@ -464,6 +457,37 @@ class ElasticQueryBuilder {
     this.query.addMust(should.length === 1 ? should[0] : { bool: { should } })
   }
 
+  /**
+  * Handle use of subject=, contributor=, title=, & callnumber= Adv Search
+  * params.
+  * Note that the RC Adv Search page may also apply filters, which are
+  * handled by `applyFilters`.
+  **/
+  applyAdvancedSearchParams () {
+    // We're specifically interested in params that match supported search-
+    // scopes because we can build a whole ES query just using the existing
+    // logic for that search scope:
+    if (this.request.advancedSearchParamsThatAreAlsoScopes()) {
```

Comment:
what are the advanced search params that are not also scopes? is that just filters?

Reply:
I realize this is older work, but I think this method wants a different name. It's not the `search_scope`ness of the params we're interested in, but rather that they're valid single value query params (as opposed to a compound `q=spaghetti&search_scope=title` format). the method itself isn't actually verifying that they are also search scopes, just that they are in the approved array of advanced search params.

Reply:
The word "advanced" in "advanced search params" is just a reminder that this is for supporting params used in RC Adv Search, but actually there are many params besides `q=` that are not filters. For example `isbn=`, `issn=`. In the future there could be more. The fact that these params are also valid `search_scope` values is actually important because we're leveraging the existing search-scope logic to build the sub-query. We very much want each of the adv search params to individually work the same as the namesake search scopes do, so we do want to use the same logic. The method thus does verify that they are also search-scopes (recent update [here](https://github.com/NYPL/discovery-api/pull/420/files#diff-deee2906e1ff785f9b725eb8ed02e9ded4ceeb86e6d790f7f0a2c781f8f1bd32R32)). In the future, there may be other Adv Search params (e.g. 'ISBN'), which would be included in an "approved array of advanced search params" but wold not have a namesake search-scope. To support that, we would have to add custom handler code to add the extra clauses for that new adv-search param. The fact that all non-filter adv search params happen to have equivalent search-scopes right now is a convenience, but not one we assume will remain, hence the awkward function name.